src/app/addPromotion/addPromotion.js

```diff
@@ -19,4 +20,27 @@ function AddPromotionComponentCtrl($exceptionHandler, $rootScope, OrderCloud, to
                 $exceptionHandler(err);
             });
     };
+}
+
+function AddRebate(OrderCloud, $rootScope, toastr) {
+    //This Service is called from the base.js on CurrentOrder
+    var service = {
+        ApplyPromo: _apply
+    };
+
+    function _apply(order) {
+        return OrderCloud.Promotions.List(null, null, null, null, null, {ValueExpression: 'order.Total * .01'})
+            .then(function(promos) {
+                if(promos.Items.length) {
+                    var promo = promos.Items[0];
+                    return OrderCloud.Orders.AddPromotion(order.ID, promo.Code)
+                        .then(function() {
+                            $rootScope.$broadcast('OC:UpdateOrder', order.ID);
+                        })
```

Comment:
I don't think this is the best solution here because we are limiting ourselves in terms of what promotion can be used in the future. 

We should pass in a specific ID and then have that ID live somewhere, possibly the catalog or if its a buyer specific promotion we can have it live on the buyer. 

By doing it this way, we can simply update the catalog with a new ID. Or if perhaps down the road when they decide they want to make it a 15% rebate we can just page the associated id